src/hooks/useApproveConfirmTransaction.ts

```diff
@@ -118,20 +122,22 @@ const useApproveConfirmTransaction = ({
         if (receipt.status) {
           dispatch({ type: 'approve_receipt' })
           onApproveSuccess({ state, receipt })
+          if (showSuccessToast) toastSuccess(t('Transaction has succeeded!'))
         }
       } catch (error) {
         dispatch({ type: 'approve_error' })
         toastError(t('Error'), t('Please try again. Confirm the transaction and make sure you are paying enough gas!'))
       }
     },
-    handleConfirm: async () => {
+    handleConfirm: async (params = {}) => {
       dispatch({ type: 'confirm_sending' })
       try {
-        const tx = await onConfirm()
+        const tx = await onConfirm(params)
         const receipt = await tx.wait()
         if (receipt.status) {
           dispatch({ type: 'confirm_receipt' })
           onSuccess({ state, receipt })
+          if (showSuccessToast) toastSuccess(t('Transaction has succeeded!'))
```

Comment:
I think in other places where this hook is used we just handle toasts inside `onApproveSuccess ` and `onSuccess` handler with customised message depending on the contract. Take a look at lottery/farmAuction/nftMarket, they all use this hook.

Reply:
Thought it might be useful to have this behavior in the hook but I can extract it.